rust/crd/src/authentication.rs

```diff
@@ -425,7 +426,7 @@ async fn check_or_generate_admin_credentials(
                 Ok(false)
             }
         }
-        false => {
+        None => {
             if !auto_generate {
```

Comment:
Shouldn't `auto_generate` be respected if the `Secret` is found too? Do we even need to fail here in that case, or just let the `Pod` fail to initialize?

Reply:
> Shouldn't `auto_generate` be respected if the `Secret` is found too?
Not sure I follow your meaning? Would you overwrite an existing secret with an autogenerated one?

Regarding not failing and letting the pod fail I'm with you on that, yes. Espcially since this may change if the secret is created sometime soon.

Reply:
No, but currently you do generate a password if a secret exists but doesn't contain one, even if `!auto_generate`.